src/core/generator/from_proto.rs

```diff
@@ -33,40 +39,102 @@ impl Context {
             map_types: Default::default(),
+            source_code_info: None,
         }
     }
 
+    /// Sets source code information for preservation of comments.
+    fn with_source_code_info(mut self, source_code_info: SourceCodeInfo) -> Self {
+        self.source_code_info = Some(source_code_info);
+        self
+    }
+
+    /// Retrieves comments associated with the source code information.
+    fn get_comments(&self, path: &[i32]) -> Option<String> {
+        self.source_code_info.as_ref().and_then(|info| {
+            info.location
+                .iter()
+                .find(|loc| loc.path == path)
+                .and_then(|loc| {
+                    loc.leading_comments.as_ref().map(|c| {
+                        c.lines()
+                            .map(|line| line.trim_start_matches('*').trim())
+                            .filter(|line| !line.is_empty())
+                            .collect::<Vec<_>>()
+                            .join("\n  ")
+                    })
+                })
+        })
+    }
+
     /// Resolves the actual name and inserts the type.
     fn insert_type(mut self, name: String, ty: Type) -> Self {
         self.config.types.insert(name.to_string(), ty);
         self
     }
 
     /// Processes proto enum types.
-    fn append_enums(mut self, enums: &Vec<EnumDescriptorProto>) -> Self {
-        for enum_ in enums {
+    fn append_enums(
+        mut self,
+        enums: &[EnumDescriptorProto],
+        parent_path: &[i32],
+        is_nested: bool,
+    ) -> Self {
+        let path_builder = PathBuilder::new(parent_path);
+        for (index, enum_) in enums.iter().enumerate() {
             let enum_name = enum_.name();
 
-            let variants = enum_
-                .value
-                .iter()
-                .map(|v| GraphQLType::new(v.name()).into_enum_variant().to_string())
-                .collect::<BTreeSet<String>>();
+            let enum_type_path = if is_nested {
+                path_builder.extend(PathField::NestedEnum, index as i32)
+            } else {
+                path_builder.extend(PathField::EnumType, index as i32)
+            };
+
+            let mut variants_with_comments = BTreeSet::new();
+
+            for (value_index, v) in enum_.value.iter().enumerate() {
+                let variant_name = GraphQLType::new(v.name()).into_enum_variant().to_string();
+
+                // Path to the enum value's comments
+                let value_path = PathBuilder::new(&enum_type_path)
+                    .extend(PathField::EnumValue, value_index as i32); // 2: value field
+
+                // Get comments for the enum value
+                let comment = self.get_comments(&value_path);
+
+                // Format the variant with its comment as description
+                if let Some(comment) = comment {
+                    let variant_with_comment =
+                        format!("\"\"\n  {}\n  \"\"\n  {}", comment, variant_name);
```

Comment:
oh, well, that's a hacky way to do this =) I think it's because of the limitation in config structure and there is no way to define `description` for enum variant, right?

let's leave it as is, since it's kinda works, but please add a TODO comment with explanation